paper.md

```diff
@@ -38,89 +38,140 @@ affiliations:
+
+With such approach, data in DAISY can be independently changed while selected project in DSW remains the same (as reply in a questionnaire). Creating a document makes a snapshot of both DSW and DAISY projects using the selected document template. First, we needed to investigate how integration question in DSW can be used and how it must be extended to allow users select own projects from DAISY. Then, we had to solve how get the data from DAISY inside a DSW document template that uses [Jinja2 language](https://jinja.palletsprojects.com).
 
 ## Integration Question for DAISY Projects
 
-*TODO: Marek - motivation, describe issues with integration question*
+DSW already supports integration question since v1.6.0 (April 2019). It allows to query a REST API for type hints – as user types reply, it suggests answers provided from the API response, and user can select the reply from integration. The advantage is configurability of the integrations and the stored ID of selected item. On the other hand, it does not allow any form of user authentication towards the external API nor more complex item selection than just regular input dropdown. In Figure 2, you can see how the integration works for selecting affiliation from [Research Organization Registry (ROR)](https://ror.org).
+
+![Integration question for affiliation using ROR.org](./figures/original-integration-question.png)
+
+We tried that it is possible to create such an integration for DAISY projects. However, it has two pitfalls:
 
-*TODO: Marek - screenshot of the DSW*
+1. To access the API, secret token must be used for authorization. Everyone who wanting to have such a question (or whole Knowledge Model) in their DSW instance would need to acquire such token and place it into server configuration file.
+2. As there is no way to distinguish specific users, it must provide all (published) projects from DAISY. The natural behavior -- displaying only projects related to the user filling the questionnaire -- is not achievable.
+
+To overcome these two issues, we designed a new type of integration question described in the following subsection.
 
 ## DSW Widget Integrations
 
-*TODO: Jan - describe the widget integration question*
+Although the integration question was a good start, we needed some improvements. The main issue was that the standard integration question in DSW connects to a public API of an external service to find the results. However, in DAISY, users need to log in first to access their projects. So the standard solution was not suitable unless all of the projects in DAISY were public, which would not be desirable.
+
+Therefore, we came up with the idea of an integration widget. Instead of a search field in DSW, we can open a popup window with a web page implemented on the connected service side, in this case, DAISY. The implementation of the widget itself is the responsibility of the connected service. That means that any necessary functionality, such as login, can be implemented.
```

Comment:
Therefore, we came up with the idea of an integration widget. Instead of a search field in DSW, we open a popup window with a web page implemented on the connected service side, in this case, DAISY. The implementation of the widget itself is the responsibility of the connected service. This means that any necessary functionality, such as login, can be implemented.

Reply:
bravo! again, it is the first time I understand what you have being doing

Reply:
```suggestion
Therefore, we came up with the idea of an integration widget. Instead of a search field in DSW, we open a popup window with a web page implemented on the connected service side, in this case, DAISY. The implementation of the widget itself is the responsibility of the connected service. This means that any necessary functionality, such as login, can be implemented.
```